### PR TITLE
feat(telegram): pass arbitrary attachments through to the harness

### PR DESCRIPTION
**Pull request by andrewagrahamhodges**
> Was originally #65, stacked on #64. #64 merged, GitHub auto-closed #65 when the base branch was deleted. Rebased onto main and reopened here.
>
> Reviewed in #65 (review thread: https://github.com/phantomyard/phantombot/pull/65).

## Summary

Phantombot was silently dropping every non-voice attachment (photos, documents, audio files, video, stickers, GIFs) because the parser only knew about \`text\` and \`voice\` fields. From the user's POV the message simply never arrived.

This PR is a deliberate **dumb pipe** — no type-aware preprocessing or content inspection. We download whatever Telegram delivers, save it under \`~/.local/share/phantombot/inbox/<chatId>/<msgId>-<filename>\`, and rewrite the user-facing message to:

\`\`\`
<caption-if-any>

[attached: <absolute-path>]
\`\`\`

The harness decides what to do — \`read\` the file (vision for images, text for docs/code/logs), ignore it, or ask the user for clarification. Phantombot never inspects, transforms, or summarizes the bytes. Voice keeps its existing dedicated STT path.

### Implementation

- **Parser**: walks every known attachment field (\`document\`, \`photo[]\`, \`audio\`, \`video\`, \`video_note\`, \`animation\`, \`sticker\`), extracts \`file_id\` + optional \`file_name\`, \`mime_type\`, \`file_size\`. For photos, picks the largest size by \`file_size\` (or \`width*height\` if size missing).
- **Filename**: Telegram's \`file_name\` if present; else \`<msg_id>-<kind>.<ext-from-mime>\`. Generic fallback for unknown MIMEs.
- **Per-chat inbox**: \`xdgDataHome()/phantombot/inbox/<chatId>/\`.
- **Edge cases**: >20 MB → skip download, surface size in the bracketed line. Download error → surface error message. No caption → just the bracketed line. Slash-command path unaffected.

### Live validation on Lena's box

\`\`\`
telegram: incoming   attachment={kind:"photo", fileName:"2201-photo.jpg", fileSize:55778, mimeType:"image/jpeg"}, captionLength:23
telegram: attachment saved   path=/home/lena/.local/share/phantombot/inbox/7995070089/918260260-2201-photo.jpg
orchestrator: trying harness gemini
telegram: complete   durationMs=56624  replyChars=848  progressEvents=13  harness=gemini  ok=true
\`\`\`

The actual gemini argv was:
\`\`\`
gemini -p "Ok, can you see it now?  [attached: /home/lena/.local/share/phantombot/inbox/7995070089/918260260-2201-photo.jpg]" -o stream-json -y
\`\`\`

56s end-to-end, 848-char reply, 13 tool events, no rate-limit retries.

## Test plan

- [x] \`bun run typecheck\` clean
- [x] \`bun test\` — 539 pass / 0 fail (was 521 on main)
- [x] New tests cover: parser (document, photo with multiple sizes, attachment-with-text fallback, voice routed separately), \`extensionFromMime\` (known/unknown/parameterized/empty), \`extractAttachment\` (synthesized filename, text-only returns undefined), \`formatAttachmentUserText\` (caption+saved, no-caption+saved, oversize, download-error), and 4 \`runTelegramServer\` integration tests (document download, oversize skip, download failure surface, no-caption case).
- [x] Live: photo with caption → fetched, saved, gemini took the turn cleanly.
- [ ] Try a PDF, a code file, a >20 MB upload — should all behave per the design.

### CODE REVIEW

**Review by lenaparkhodges — DISMISSED** [review]
## Code Review: feat(telegram-attachments)

**Author:** @andrewagrahamhodges | **Files:** 2 | **Verdict:** Approve

This PR transforms Phantombot from a text+voice-only bot into a versatile "dumb pipe" for any file Telegram can deliver. The implementation is clean, well-scoped, and covered by a robust suite of unit and integration tests.

### Findings

- **Dumb Pipe Philosophy:** Letting the harness decide how to handle files (read/vision/ignore) is the right architectural choice for Phantombot.
- **Robustness:** Handles oversize files (>20 MB cap) and download failures by informing the harness rather than crashing or swallowing the event.
- **Clean Parser:** The logic to pick the best photo size and synthesize filenames for extension-less attachments is solid.
- **Test Coverage:** Excellent integration tests covering download successes, failures, and edge cases like oversize files.

<details>
<summary>📋 AI Fix Prompt (copy to your AI editor)</summary>

```
Review and approve PR #66 (feat(telegram): pass arbitrary attachments through to the harness).

The changes are well-implemented and fully tested.
```
</details>

**Review by lenaparkhodges — COMMENTED** [review]
## Code Review: feat(telegram): attachments pipe

**Author:** @andrewagrahamhodges | **Files:** 2 | **Verdict:** Comment Only

This is a great addition that solves the "black hole" problem for Telegram attachments. The approach of a "dumb pipe" that leaves content inspection to the harness is consistent with the Phantombot design philosophy.

### Findings

| # | Severity | File | Line | Finding |
|---|----------|------|------|---------|
| 1 | 🟠 Major | src/channels/telegram.ts | 822 | Path traversal risk via `att.fileName`. |
| 2 | 🔵 Trivial | src/channels/telegram.ts | 338 | Photo size ranking is robust. |

### What's Good
- Excellent test coverage including edge cases like oversized files and download failures.
- The MIME to extension mapping is pragmatic and handles weird types gracefully.
- Using the per-chat inbox directory ensures logical separation of assets.

<details>
<summary>📋 AI Fix Prompt (copy to your AI editor)</summary>

```
Review the findings in PR #66 (feat(telegram): attachments pipe).

1. [Major] src/channels/telegram.ts:822 — Sanitize `att.fileName` to prevent path traversal. If a document has a filename like "../../etc/passwd", `path.join` might resolve it outside the inbox directory. Fix by wrapping the filename in `path.basename()` before joining.

Repository: phantomyard/phantombot
Branch: feat/telegram-attachments
Base: main
```
</details>

File: src/channels/telegram.ts [hunk 1]
```diff
@@ -527,6 +796,62 @@ async function processChatMessage(
...
+      oversizeBytes = att.fileSize;
+      log.warn("telegram: attachment over bot-API cap, not downloading", {
+        chatId: msg.chatId,
+        kind: att.kind,
+        fileName: att.fileName,
+        fileSize: att.fileSize,
+      });
+    } else {
+      try {
+        const dir = inboxDir(msg.chatId);
```
Comment (lenaparkhodges) [inline src/channels/telegram.ts:822]: 🟠 **Major**: Potential path traversal vulnerability. If `att.fileName` (which comes from Telegram's `file_name` field for documents) contains path separators or directory climbing characters (e.g., `../../evil.txt`), `join` might resolve the path outside of the intended inbox directory. 

**Fix**: Import `basename` from `node:path` and use it to wrap the filename: 
`const path = join(dir, `${msg.updateId}-${basename(att.fileName)}`);`

File: src/channels/telegram.ts [hunk 2]
```diff
@@ -216,20 +297,177 @@ function abortReasonString(reason: unknown): string {
...
+    video?: TelegramRawFile;
+    video_note?: TelegramRawFile;
+    animation?: TelegramRawFile;
+    sticker?: TelegramRawFile;
+    photo?: TelegramRawPhotoSize[];
   };
 }
 
+/**
+ * Walk a raw message looking for an attachment. Photos are an array of
```
Comment (lenaparkhodges) [inline src/channels/telegram.ts:338]: 🔵 **Trivial**: Nice work on the photo size ranking logic. Falling back to `width * height` when `file_size` is missing ensures we always get the best available quality.

File: src/channels/telegram.ts (hunk 1 above)
Comment (andrewagrahamhodges) [reply src/channels/telegram.ts:822]: Fixed in 7a61b0f — wrapped `att.fileName` in `basename()` at the path-construction site, plus a regression test that verifies `file_name="../../etc/passwd"` lands inside the inbox dir as `559-passwd`.

File: src/channels/telegram.ts [hunk 3]
```diff
@@ -527,6 +796,66 @@ async function processChatMessage(
...
+
+    if (
+      att.fileSize !== undefined &&
+      att.fileSize > TELEGRAM_BOT_DOWNLOAD_CAP_BYTES
+    ) {
+      oversizeBytes = att.fileSize;
+      log.warn("telegram: attachment over bot-API cap, not downloading", {
+        chatId: msg.chatId,
+        kind: att.kind,
+        fileName: att.fileName,
```
Comment (andrewagrahamhodges) [inline src/channels/telegram.ts:817]: 🔵 **Trivial**: Verified the `basename(att.fileName)` fix. This effectively prevents any directory climbing via Telegram's `file_name` field.

File: tests/channels-telegram.test.ts [hunk 4]
```diff
@@ -189,6 +404,225 @@ const baseConfig = (
...
+    const transport = new FakeTransport();
+    transport.fileResponses.set("flaky", new Error("ECONNRESET"));
+    transport.pendingUpdates.push({
+      updateId: 557,
+      chatId: 1001,
+      fromUserId: 42,
+      text: "",
+      attachment: {
+        fileId: "flaky",
+        fileName: "fail.png",
```
Comment (andrewagrahamhodges) [inline tests/channels-telegram.test.ts:517]: 🔵 **Trivial**: The new regression test `malicious file_name with path separators is contained inside the inbox dir` correctly verifies the fix and protects against future regressions.

**Comment by andrewagrahamhodges** [comment]
## Code Review: feat(telegram): attachments pipe (Self-Review Update)

**Author:** @andrewagrahamhodges | **Files:** 2 | **Verdict:** Approve

Thank you for addressing the path traversal vulnerability. The addition of `basename()` sanitization and the corresponding regression test makes this feature ready for merge. This is a very robust implementation of the attachment pipe.

### Findings

| # | Severity | File | Line | Finding |
|---|----------|------|------|---------|
| 1 | 🔵 Trivial | src/channels/telegram.ts | 817 | Path traversal vulnerability is now closed. |
| 2 | 🔵 Trivial | tests/channels-telegram.test.ts | 517 | Excellent regression test for path traversal. |

### What's Good
- **Security:** Promptly addressed the path traversal risk with a clean fix.
- **Testing:** Comprehensive test suite covers functional requirements, edge cases, and security regressions.
- **Architecture:** The "dumb pipe" approach is correctly implemented and maintains project simplicity.

<details>
<summary>📋 AI Fix Prompt (copy to your AI editor)</summary>

```
Review and approve PR #66 (feat(telegram): attachments pipe).

The path traversal vulnerability has been fixed and verified with a regression test.
```
</details>

**Review by lenaparkhodges — APPROVED** [review]
Official approval following fix and verification of the path traversal vulnerability. LGTM.
